### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by ShikhaChauhan14**

### CODE REVIEW

File: src/server/handleMsg.cc [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#include "./server.h"
+
+void handleMsg(const int currentClientFd, ChatroomToFdList &chatRooms, FdToName &clients, string msg) {
```
Comment (sidvishnoi) [inline src/server/handleMsg.cc:3]: you forgot to call this in `serverChat.cc`. oops.

**Review by sidvishnoi — CHANGES_REQUESTED** [review]
you accidentally forgot to call `handleMsg` in `serverChat.cc`. Rest is good.

File: src/server/handleLostConnection.cc [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  logout(db, username);
+  clients.erase(currentClientFd);
+  string info = "INFO" + DELIM + username + DELIM + "went offline";
+  cout << color::magenta << "[CLIENT:OFFLINE] " << color::reset
+    << "<" << username << "> went offline" << endl;
+  
+  for (auto &room : chatRooms) {
+    auto members = room.second;
+    auto receiverItr = members.find(currentClientFd);
+    if (receiverItr != members.end()) {
```
Comment (sidvishnoi) [inline src/server/handleLostConnection.cc]: maybe replace with

``` cpp
if (members.find(currentClientFd) != members.end()) {
```

File: src/server/handleMsg.cc [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      if (index != string::npos) {
+        std::string receiverName = text.substr(index + 1, text.substr(index + 1).find(' '));
+        auto receiverItr = find_if(
+          clients.begin(),
+          clients.end(),
+          [&receiverName](auto const &itr) -> bool {
+            return itr.second == receiverName;
+          }
+        );
+        if (receiverItr != clients.end() && chatRooms.at(chatRoomName).count(receiverItr->first) != 0) {
```
Comment (sidvishnoi) [inline src/server/handleMsg.cc]: add a comment here what it does

Comment (ShikhaChauhan14) [reply src/server/handleMsg.cc]: ok

File: src/server/handleMsg.cc (hunk 1 above)
Comment (ShikhaChauhan14) [reply src/server/handleMsg.cc:3]: Thanks

File: src/server/handleLostConnection.cc (hunk 2 above)
Comment (ShikhaChauhan14) [reply src/server/handleLostConnection.cc]: Yes, Line 20 and 21 can be merged.

**Pull request merged**